projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts

```diff
@@ -1,68 +1,137 @@
-import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
-import { TsFilterItem, FilterBarItemType } from './model/ts-filter-item';
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
+} from '@angular/core';
+import { debounceTime, Subject, take } from 'rxjs';
 import { TimeSeriesContext } from '../../time-series-context';
-import { debounce, debounceTime, Subject } from 'rxjs';
 import { FilterUtils } from '../../util/filter-utils';
+import { PerformanceViewSettings } from '../model/performance-view-settings';
+import { PerformanceViewTimeSelectionComponent } from '../time-selection/performance-view-time-selection.component';
+import { FilterBarItemComponent } from './item/filter-bar-item.component';
+import { FilterBarItemType, TsFilterItem } from './model/ts-filter-item';
 
 @Component({
   selector: 'step-ts-filter-bar',
   templateUrl: './filter-bar.component.html',
   styleUrls: ['./filter-bar.component.scss'],
 })
 export class FilterBarComponent implements OnInit, OnDestroy {
-  readonly EMIT_DEBOUNCE_TIME = 300;
   @Input() context!: TimeSeriesContext;
+  @Input() performanceViewSettings!: PerformanceViewSettings;
```

Comment:
such architectural improvements could be done at a later stage as this is urgent for the release and having gone through the components involved I can say that there would be some other ones where we could do such improvements as well, so I reckon it would be best to create a separate ticket specifically for that, discuss the use cases for these components and decide on the best way to organize them